azkaban-exec-server/src/main/java/azkaban/dag/Node.java

```diff
@@ -39,35 +39,34 @@
 
   private Status status = Status.READY;
 
-  private Dag dag;
+  private final Dag dag;
 
-  Node(final String name, final NodeProcessor nodeProcessor) {
-    this.name = name;
+  Node(final String name, final NodeProcessor nodeProcessor, final Dag dag) {
     requireNonNull(nodeProcessor, "The nodeProcessor parameter can't be null.");
     this.nodeProcessor = nodeProcessor;
+    requireNonNull(name, "The name of the node can't be null");
+    this.name = name;
+    requireNonNull(name, "The dag of the node can't be null");
```

Comment:
small type - I think `name` is supposed to be `dag`, right?

Reply:
Thanks. Will fix.

Reply:
Fixed in the next patch.